app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala

```diff
@@ -75,7 +76,17 @@ class BindingTariffClassificationConnector @Inject()(appConfig: AppConfig, clien
   }
 
   def findEvents(reference: String, pagination: Pagination)(implicit hc: HeaderCarrier): Future[Paged[Event]] = {
-    val url = s"${appConfig.bindingTariffClassificationUrl}/cases/$reference/events?page=${pagination.page}&page_size=${pagination.pageSize}"
+    findFilteredEvents(reference,pagination, None)
+  }
+
+  def findFilteredEvents(reference: String, pagination: Pagination, onlyEventTypes: Option[Set[EventType]])(implicit hc: HeaderCarrier): Future[Paged[Event]] = {
+
+    val searchParam = onlyEventTypes match {
+      case Some(only) => s"case_reference=$reference${only.mkString("&type=","&type=","&")}"
+      case None => s"case_reference=$reference&"
```

Comment:
probably this last & should be on the rest call . easy peasy

Reply:
Changed this alongside simplification changes above.